File: LeetCode/Problems/InvertBinaryTree.cs
using LeetCode.Structures;

namespace LeetCode.Problems
{
    public class InvertBinaryTree
    {
        public TreeNode InvertTree(TreeNode root)
        {
            if (root == null)
                return null;
            
            InvertTree(root.left);
            InvertTree(root.right);

            var tempRight = root.right;
            root.right = root.left;
            root.left = tempRight;

            return root;
        }
    }
}

File: LeetCode/Python/Trees/InvertBinaryTree.py
from typing import Optional
from Structures.TreeNode import BinaryTreeNode


class Solution:

    def invertTree(self, root: Optional[BinaryTreeNode]) -> Optional[BinaryTreeNode]:

        if root is None:
            return None
        
        #swap
        temp = root.left
        root.left = root.right
        root.right = temp 
        
        # invert children
        self.invertTree(root.left)
        self.invertTree(root.right)
        
        return root